portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -30,47 +36,71 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+
 /** Servlet that returns comments and updates the datastore with comments*/
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  
+    
+  private final Gson gson = new Gson();
   DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  UserService userService = UserServiceFactory.getUserService();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
     int numComments = Integer.valueOf(request.getParameter("numComments"));
-    Query query = new Query("Comment"); // get by date posted
-
+    String language = request.getParameter("language");
+    // sort by time posted
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING); 
     PreparedQuery results = datastore.prepare(query);
 
     List<Comment> comments= new ArrayList<>();
+
+    Translate translate = TranslateOptions.getDefaultInstance().getService();
     
     for (Entity entity: results.asIterable()){
         if(comments.size() < numComments){
           long id = entity.getKey().getId();
           String text = (String) entity.getProperty("text");
-          long timestamp = (long) entity.getProperty("timestamp");
 
-          Comment comment = new Comment(id,text,timestamp);
+          // Do the translation.
+          Translation translation =
+            translate.translate(text, Translate.TranslateOption.targetLanguage(language));
```

Comment:
If the target language is always the same, you could put the TranslateOption in a pre-loop variable as well.